From 95955e6746e847a366143121a8d0a61404cc740e Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Wed, 8 Jul 2020 08:05:06 -0700
Subject: [PATCH 13/13] factor: treat ' +bignum' like non-bignum

* src/factor.c (strto2uintmax): Instead of here ...
(print_factors): ... skip spaces and '+' here, so that
bignums are treated like non-bignums.
* tests/misc/factor.pl (bug-gmp-plus_2_sup_128_plus_1): New test.
---
 tests/misc/factor.pl | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/misc/factor.pl b/tests/misc/factor.pl
index ecd2ab293..db64706ab 100755
--- a/tests/misc/factor.pl
+++ b/tests/misc/factor.pl
@@ -86,6 +86,9 @@ my @Tests =
       '115792089237316195423570985008687907853'
       . '269984665640564039457584007913129639936',
       {OUT => '2 'x255 . '2'}],
+     ['bug-gmp-plus_2_sup_128_plus_1',
+      '+170141183460469231731687303715884105729',
+      {OUT => '3 56713727820156410577229101238628035243'}],
     );
 
 
-- 
2.27.0

